wolfSSL/src/chapter02.md

```diff
@@ -680,15 +1234,325 @@ Enable ECC SECP384R1 Single-Precision (SP) support. Only applies to `WOLFSSL_SP_
+
+#### WOLFSSL_DTLS_CH_FRAG
+
+Allows a server to process a fragmented second/verified (one containing a valid cookie response) ClientHello message. The first/unverified (one without a cookie extension) ClientHello MUST be unfragmented so that the DTLS server can process it statelessly. This is only implemented for DTLS 1.3. The user MUST call wolfSSL_dtls13_allow_ch_frag() on the server to explicitly enable this during runtime. "Using DTLS 1.3 + pqc without WOLFSSL_DTLS_CH_FRAG will probably fail. Use --enable-dtls-frag-ch to enable it."
+
+#### WOLFSSL_DTLS_MTU_ADDITIONAL_READ_BUFFER
+
+We need additional bytes to read so that we can work with a peer who has a slightly different MTU than us.
+
+#### WOLFSSL_DTLS_WINDOW_WORDS
+
+Used to check storage size or to verify if an index is valid for window.
+
+#### WOLFSSL_EXPORT_SPC_SZ
+
+Define to specify the amount of bytes used from CipherSpecs.
+
+#### WOLFSSL_MIN_DOWNGRADE
+
+Specifies minimum downgrade version.
+
+#### WOLFSSL_MIN_DTLS_DOWNGRADE
+
+Specifies minimum DTLS downgrade version.
+
+#### WOLFSSL_MIN_ECC_BITS
+
+Can set minimum ECC key size allowed.
+
+#### WOLFSSL_MIN_RSA_BITS
+
+By default, wolfSSL restricts RSA key sizes to 1024-bits minimum. To allow the decoding of smaller, less secure RSA keys like 512-bit keys, you will need to add the compiler flag `-DWOLFSSL_MIN_RSA_BITS=512` to CFLAGS or CPPFLAGS or define it in your user-settings header.
+
+#### WOLFSSL_MODE_AUTO_RETRY_ATTEMPTS
+
+Used to limit the possibility of an infinite retry loop.
+
+#### WOLFSSL_MULTICAST
+
+DTLS multicast feature.
+
+#### WOLFSSL_MULTICAST_PEERS
+
+Multicast feature defined as max allowed 100 peers.
+
+#### WOLFSSL_NAMES_STATIC
+
+Uses static ECC structs for Position Independent Code (PIC).
+
+#### WOLFSSL_SEND_HRR_COOKIE
+
+TLS extension used by DTLS 1.3.
+
+#### WOLFSSL_SEP
+
+Feature certificate policy set extension.
+
+#### WOLFSSL_SESSION_ID_CTX
+
+Used to copy over application session context ID.
+
+#### WOLFSSL_SESSION_TIMEOUT
+
+Default session resumption cache timeout in seconds is used to define timeout manually.
+
+#### KEEP_OUR_CERT
+
+Used to ensure the ability to return SSL certificate.
+
+#### KEEP_PEER_CERT
+
+Retains peer certs. Parts of the OpenSSL compatibility layer require peer certs.
+
+#### WOLFSSL_SIGNER_DER_CERT
+
+Used to no CERT DER buffer OR NULL cm. Used to allocate DER and cert.
+
+#### CA_TABLE_SIZE
+
+wolfSSL certificate manager. CA signer table. Used in certificate manager and be defined size.
+
+#### ECDHE_SIZE
+
+Define to allow this to be overridden at compile-time. ECDHE server size defaults to 256 bits, which can set a predetermined ECDHE curve size. The default is 32 bytes.
+
+#### CIPHER_NONCE
+
+It is used as a cryptographic number, which is implemented in authentication. It is a pseudo-random number, which is an integrity-only cipher suite.
+
+#### WOLFSSL_USE_POPEN_HOST
+
+Used to implement popen and "host".
```

Comment:
Uses `popen`for creating socket with host and post for wolfio.c socket open code used with CRL and OCSP.